lib/semantics/check-allocate.cc

```diff
@@ -620,6 +620,7 @@ bool AllocationCheckerHelper::RunCoarrayRelatedChecks(
         "Allocatable object must not be coindexed in ALLOCATE"_err_en_US);
     return false;
   }
+  context.CheckDoVarRedefine(name_);
```

Comment:
A minor point: this check is not related to coarrays yet is is done in `RunCoarrayRelatedChecks`. To avoid misleading future maintainers, I would move this check just before the following line instead: https://github.com/flang-compiler/f18/blob/867dd045de033bd092ae476a852e57f3b4bd3213/lib/semantics/check-allocate.cc#L542

Reply:
Thanks, Jean.  That's a better place.  I'll move it.